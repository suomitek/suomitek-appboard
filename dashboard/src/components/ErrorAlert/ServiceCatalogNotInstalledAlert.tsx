import * as React from "react";

import NotFoundErrorAlert from "./NotFoundErrorAlert";

class ServiceCatalogNotInstalledAlert extends React.Component {
  public render() {
    return (
      <NotFoundErrorAlert header="Service Catalog not installed.">
        <div>
          <p>
            Ask an administrator to install the{" "}
            <a
              href="https://github.com/kubernetes-incubator/service-catalog"
              target="_blank"
              rel="noopener noreferrer"
            >
              Kubernetes Service Catalog
            </a>{" "}
            to browse, provision and manage external services within Suomitek-appboard.
          </p>
        </div>
      </NotFoundErrorAlert>
    );
  }
}

export default ServiceCatalogNotInstalledAlert;
