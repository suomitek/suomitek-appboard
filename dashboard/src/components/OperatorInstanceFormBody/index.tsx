import OperatorInstanceFormBody from "./OperatorInstanceFormBody";

export default OperatorInstanceFormBody;
