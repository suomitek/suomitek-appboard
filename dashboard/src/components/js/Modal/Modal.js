import React from "react";
import PropTypes from "prop-types";

import { get } from "lodash-es";

export const modalSizes = {
  sm: "modal-sm",
  md: "modal-md",
  lg: "modal-lg",
  xl: "modal-xl",
};

const Modal = ({
  title,
  children,
  footer,
  hideCloseButton,
  modalSize,
  onModalClose,
  showModal,
  staticBackdrop,
}) => {
  const onClose = () => onModalClose();

  if (!showModal) {
    return null;
  }

  const size = get(modalSizes, modalSize, modalSizes["md"]);
  const closeOnBackdrop = staticBackdrop ? () => {} : onModalClose;

  return (
    <div className="modal">
      <div
        className={`modal-dialog ${size}`}
        aria-modal={true}
        role="dialog"
        aria-hidden={showModal}
      >
        <div className="modal-content">
          <div className="modal-header">
            {!hideCloseButton && (
              <button
                aria-label="Close the current modal"
                className="close"
                type="button"
                onClick={onClose}
              >
                <clr-icon aria-hidden="true" shape="close" />
              </button>
            )}
            <h3 className="modal-title">{title}</h3>
          </div>
          <div className="modal-body">{children}</div>
          <div className="modal-footer">{footer}</div>
        </div>
      </div>
      <div onClick={closeOnBackdrop} className="modal-backdrop" aria-hidden="true" />
    </div>
  );
};

Modal.defaultProps = {
  modalSize: "md",
  hideCloseButton: false,
  showModal: false,
  staticBackdrop: true,
};

Modal.propTypes = {
  title: PropTypes.string,
  children: PropTypes.node.isRequired,
  footer: PropTypes.node,
  hideCloseButton: PropTypes.bool.isRequired,
  modalSize: PropTypes.string.isRequired,
  onModalClose: PropTypes.func.isRequired,
  showModal: PropTypes.bool.isRequired,
  staticBackdrop: PropTypes.bool,
};

export default Modal;
