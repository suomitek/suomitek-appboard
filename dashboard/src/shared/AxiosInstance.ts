import Axios, { AxiosError, AxiosInstance, AxiosRequestConfig, AxiosResponse } from "axios";

import { Action, Store } from "redux";
import { ThunkDispatch } from "redux-thunk";
import actions from "../actions";
import { Auth } from "./Auth";
import {
  ConflictError,
  ForbiddenError,
  InternalServerError,
  IStoreState,
  NotFoundError,
  UnauthorizedError,
  UnprocessableEntity,
} from "./types";

export function addAuthHeaders(axiosInstance: AxiosInstance) {
  axiosInstance.interceptors.request.use((config: AxiosRequestConfig) => {
    const authToken = Auth.getAuthToken();
    if (authToken) {
      config.headers.Authorization = `Bearer ${authToken}`;
    }
    return config;
  });
}

export function addErrorHandling(axiosInstance: AxiosInstance, store: Store<IStoreState>) {
  axiosInstance.interceptors.response.use(
    response => response,
    e => {
      const dispatch = store.dispatch as ThunkDispatch<IStoreState, null, Action>;
      const err: AxiosError = e;
      if (
        err.code === undefined &&
        err.message === "Network Error" &&
        !err.response &&
        Auth.usingOIDCToken()
      ) {
        // The OIDC token is no longer valid, logout
        dispatch(actions.auth.expireSession());
      }
      let message = err.message;
      if (err.response) {
        if (err.response.data.message) {
          message = err.response.data.message;
        }
        if (typeof err.response.data === "string") {
          message = err.response.data;
        }
      }

      const dispatchErrorAndLogout = (m: string) => {
        // Global action dispatch to log the user out
        dispatch(actions.auth.authenticationError(m));
        // Expire the session if we are using OIDC tokens and
        // logout either way.
        dispatch(actions.auth.expireSession());
      };
      const response = err.response as AxiosResponse;
      switch (response && response.status) {
        case 401:
          dispatchErrorAndLogout(message);
          return Promise.reject(new UnauthorizedError(message));
        case 403:
          // A 403 directly from the auth proxy requires reauthentication.
          if (Auth.usingOIDCToken() && Auth.is403FromAuthProxy(response)) {
            dispatchErrorAndLogout(message);
          }
          return Promise.reject(new ForbiddenError(message));
        case 404:
          return Promise.reject(new NotFoundError(message));
        case 409:
          return Promise.reject(new ConflictError(message));
        case 422:
          return Promise.reject(new UnprocessableEntity(message));
        case 500:
          return Promise.reject(new InternalServerError(message));
        default:
          return Promise.reject(new Error(message));
      }
    },
  );
}

// Error handling is added with an interceptor in index.tsx
export const axios = Axios.create();
// Authorization headers and error handling are added with an interceptor in index.tsx
export const axiosWithAuth = Axios.create();
