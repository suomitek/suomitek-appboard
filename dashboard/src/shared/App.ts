import * as url from "shared/url";
import { axiosWithAuth } from "./AxiosInstance";
import { hapi } from "./hapi/release";
import { IAppOverview, IChartVersion } from "./types";

export const KUBEOPS_ROOT_URL = "api/tiller-deploy/v1";

export class App {
  public static async create(
    cluster: string,
    namespace: string,
    releaseName: string,
    chartNamespace: string,
    chartVersion: IChartVersion,
    values?: string,
  ) {
    const chartAttrs = chartVersion.relationships.chart.data;
    const endpoint = url.kubeops.releases.list(cluster, namespace);
    const { data } = await axiosWithAuth.post(endpoint, {
      appRepositoryResourceName: chartAttrs.repo.name,
      appRepositoryResourceNamespace: chartNamespace,
      chartName: chartAttrs.name,
      releaseName,
      values,
      version: chartVersion.attributes.version,
    });
    return data;
  }

  public static async upgrade(
    cluster: string,
    namespace: string,
    releaseName: string,
    chartNamespace: string,
    chartVersion: IChartVersion,
    values?: string,
  ) {
    const chartAttrs = chartVersion.relationships.chart.data;
    const endpoint = url.kubeops.releases.get(cluster, namespace, releaseName);
    const { data } = await axiosWithAuth.put(endpoint, {
      appRepositoryResourceName: chartAttrs.repo.name,
      appRepositoryResourceNamespace: chartNamespace,
      chartName: chartAttrs.name,
      releaseName,
      values,
      version: chartVersion.attributes.version,
    });
    return data;
  }

  public static async rollback(
    cluster: string,
    namespace: string,
    releaseName: string,
    revision: number,
  ) {
    const endpoint = url.kubeops.releases.get(cluster, namespace, releaseName);
    const { data } = await axiosWithAuth.put(
      endpoint,
      {},
      {
        params: {
          action: "rollback",
          revision,
        },
      },
    );
    return data;
  }

  public static async delete(
    cluster: string,
    namespace: string,
    releaseName: string,
    purge: boolean,
  ) {
    let endpoint = url.kubeops.releases.get(cluster, namespace, releaseName);
    if (purge) {
      endpoint += "?purge=true";
    }
    const { data } = await axiosWithAuth.delete(endpoint);
    return data;
  }

  public static async listApps(cluster: string, namespace?: string, allStatuses?: boolean) {
    let endpoint = namespace
      ? url.kubeops.releases.list(cluster, namespace)
      : url.kubeops.releases.listAll(cluster);
    if (allStatuses) {
      endpoint += "?statuses=all";
    }
    const { data } = await axiosWithAuth.get<{ data: IAppOverview[] }>(endpoint);
    return data.data;
  }

  public static async getRelease(cluster: string, namespace: string, name: string) {
    const { data } = await axiosWithAuth.get<{ data: hapi.release.Release }>(
      url.kubeops.releases.get(cluster, namespace, name),
    );
    return data.data;
  }
}
