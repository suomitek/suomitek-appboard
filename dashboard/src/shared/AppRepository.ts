import { axiosWithAuth } from "./AxiosInstance";
import { APIBase } from "./Kube";
import { ICreateAppRepositoryResponse } from "./types";
import * as url from "./url";

export class AppRepository {
  public static async list(namespace: string) {
    const { data } = await axiosWithAuth.get(AppRepository.getSelfLink(namespace));
    return data;
  }

  public static async get(name: string, namespace: string) {
    const { data } = await axiosWithAuth.get(AppRepository.getSelfLink(namespace, name));
    return data;
  }

  public static async resync(name: string, namespace: string) {
    const repo = await AppRepository.get(name, namespace);
    repo.spec.resyncRequests = repo.spec.resyncRequests || 0;
    repo.spec.resyncRequests++;
    const { data } = await axiosWithAuth.put(AppRepository.getSelfLink(namespace, name), repo);
    return data;
  }

  public static async update(
    name: string,
    namespace: string,
    repoURL: string,
    authHeader: string,
    customCA: string,
    syncJobPodTemplate: any,
    registrySecrets: string[],
  ) {
    const { data } = await axiosWithAuth.put<ICreateAppRepositoryResponse>(
      url.backend.apprepositories.update(namespace, name),
      {
        appRepository: { name, repoURL, authHeader, customCA, syncJobPodTemplate, registrySecrets },
      },
    );
    return data;
  }

  public static async delete(name: string, namespace: string) {
    const { data } = await axiosWithAuth.delete(
      url.backend.apprepositories.delete(name, namespace),
    );
    return data;
  }

  // create uses the kubeapps backend API
  // TODO(mnelson) Update other endpoints to similarly use the backend API, removing the need
  // for direct k8s api access (for this resource, at least).
  public static async create(
    name: string,
    namespace: string,
    repoURL: string,
    authHeader: string,
    customCA: string,
    syncJobPodTemplate: any,
    registrySecrets: string[],
  ) {
    const { data } = await axiosWithAuth.post<ICreateAppRepositoryResponse>(
      url.backend.apprepositories.create(namespace),
      {
        appRepository: { name, repoURL, authHeader, customCA, syncJobPodTemplate, registrySecrets },
      },
    );
    return data;
  }

  public static async validate(repoURL: string, authHeader: string, customCA: string) {
    const { data } = await axiosWithAuth.post<any>(url.backend.apprepositories.validate(), {
      appRepository: { repoURL, authHeader, customCA },
    });
    return data;
  }

  private static APIBase: string = APIBase;
  private static APIEndpoint: string = `${AppRepository.APIBase}/apis/kubeapps.com/v1alpha1`;
  private static getSelfLink(namespace: string, name?: string): string {
    return `${AppRepository.APIEndpoint}/namespaces/${namespace}/apprepositories${
      name ? `/${name}` : ""
    }`;
  }
}
