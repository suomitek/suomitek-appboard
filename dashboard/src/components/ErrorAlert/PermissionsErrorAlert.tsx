import * as React from "react";
import { Lock } from "react-feather";

import { UnexpectedErrorAlert } from ".";
import { IRBACRole } from "../../shared/types";
import { namespaceText } from "./helpers";
import PermissionsListItem from "./PermissionsListItem";

interface IPermissionsErrorPage {
  action: string;
  roles: IRBACRole[];
  rawMessage: string;
  namespace: string;
}

class PermissionsErrorPage extends React.Component<IPermissionsErrorPage> {
  public render() {
    const { action, namespace, roles, rawMessage } = this.props;
    return (
      <UnexpectedErrorAlert
        title={
          <span>
            You don't have sufficient permissions to {action}{" "}
            {namespace && <>in {namespaceText(namespace)}</>}
          </span>
        }
        icon={Lock}
        showGenericMessage={false}
      >
        <div>
          {roles.length > 0 ? (
            <>
              <p>Ask your administrator for the following RBAC roles:</p>
              <ul className="error__permisions-list">
                {roles.map((r, i) => (
                  <PermissionsListItem key={i} namespace={namespace} role={r} />
                ))}
              </ul>
            </>
          ) : (
            <>
              <p>The following error was returned:</p>
              <div className="error__content">
                <section className="Terminal terminal__error elevation-1 type-color-white error__text">
                  {rawMessage}
                </section>
              </div>
            </>
          )}
          <p>
            See the documentation for more info on{" "}
            <a
              href="https://github.com/kubeapps/kubeapps/blob/master/docs/user/access-control.md"
              target="_blank"
              rel="noopener noreferrer"
            >
              access control in Kubeapps
            </a>
            .
          </p>
        </div>
      </UnexpectedErrorAlert>
    );
  }
}

export default PermissionsErrorPage;
