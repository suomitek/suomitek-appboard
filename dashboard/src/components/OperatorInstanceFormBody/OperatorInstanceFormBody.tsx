import * as yaml from "js-yaml";
import { get } from "lodash";
import * as React from "react";
import { Tab, TabList, TabPanel, Tabs } from "react-tabs";

import { definedNamespaces } from "../../shared/Namespace";
import { IClusterServiceVersionCRD, IResource, UnprocessableEntity } from "../../shared/types";
import ConfirmDialog from "../ConfirmDialog";
import AdvancedDeploymentForm from "../DeploymentFormBody/AdvancedDeploymentForm";
import Differential from "../DeploymentFormBody/Differential";
import { ErrorSelector } from "../ErrorAlert";
import UnexpectedErrorPage from "../ErrorAlert/UnexpectedErrorAlert";
import LoadingWrapper from "../LoadingWrapper";

export interface IOperatorInstanceFormProps {
  csvName: string;
  isFetching: boolean;
  namespace: string;
  handleDeploy: (resource: IResource) => void;
  defaultValues: string;
  crd?: IClusterServiceVersionCRD;
  errors: {
    fetch?: Error;
    create?: Error;
    update?: Error;
  };
}

export interface IOperatorInstanceFormBodyState {
  values: string;
  restoreDefaultValuesModalIsOpen: boolean;
  submittedResourceName: string;
  error?: Error;
}

class DeploymentFormBody extends React.Component<
  IOperatorInstanceFormProps,
  IOperatorInstanceFormBodyState
> {
  public state: IOperatorInstanceFormBodyState = {
    values: this.props.defaultValues,
    submittedResourceName: "",
    restoreDefaultValuesModalIsOpen: false,
  };

  public componentDidUpdate(prevProps: IOperatorInstanceFormProps) {
    const { defaultValues } = this.props;
    if (prevProps.defaultValues !== defaultValues) {
      this.setState({ values: defaultValues });
    }
  }

  public render() {
    const { isFetching, errors, csvName, namespace, crd } = this.props;
    const { submittedResourceName, error } = this.state;

    if (errors.fetch) {
      return (
        <ErrorSelector
          error={errors.fetch}
          resource={`Operator Instance "${csvName}" (${crd?.name})`}
        />
      );
    }
    if (namespace === definedNamespaces.all) {
      return <UnexpectedErrorPage title="Select a namespace before creating a new instance." />;
    }
    if (isFetching) {
      return <LoadingWrapper />;
    }
    return (
      <>
        {errors.create && (
          <ErrorSelector
            error={errors.create}
            resource={`Operator Instance "${submittedResourceName}"`}
          />
        )}
        {errors.update && (
          <ErrorSelector
            error={errors.update}
            resource={`Operator Instance "${submittedResourceName}"`}
          />
        )}
        {error && <ErrorSelector error={error} resource={"Operator Instance"} />}
        <ConfirmDialog
          modalIsOpen={this.state.restoreDefaultValuesModalIsOpen}
          loading={false}
          confirmationText={"Are you sure you want to restore the default instance values?"}
          confirmationButtonText={"Restore"}
          onConfirm={this.restoreDefaultValues}
          closeModal={this.closeRestoreDefaultValuesModal}
        />
        {this.renderTabs()}
        <form className="container padding-b-bigger" onSubmit={this.handleDeploy}>
          <div className="margin-t-big">
            <button className="button button-primary" type="submit">
              Submit
            </button>
            <button className="button" type="button" onClick={this.openRestoreDefaultValuesModal}>
              Restore Defaults
            </button>
          </div>
        </form>
      </>
    );
  }

  private handleValuesChange = (values: string) => {
    this.setState({ values });
  };

  private renderTabs = () => {
    return (
      <div className="margin-t-normal row">
        <Tabs className="col-8">
          <TabList>
            <Tab>YAML</Tab>
            <Tab>Changes</Tab>
          </TabList>
          <TabPanel>
            <AdvancedDeploymentForm
              appValues={this.state.values}
              handleValuesChange={this.handleValuesChange}
            />
          </TabPanel>
          <TabPanel>
            <Differential
              title="Difference from example defaults"
              oldValues={this.props.defaultValues}
              newValues={this.state.values}
              emptyDiffText="No changes detected from example defaults"
            />
          </TabPanel>
        </Tabs>
      </div>
    );
  };

  private closeRestoreDefaultValuesModal = () => {
    this.setState({ restoreDefaultValuesModalIsOpen: false });
  };

  private openRestoreDefaultValuesModal = () => {
    this.setState({ restoreDefaultValuesModalIsOpen: true });
  };

  private restoreDefaultValues = () => {
    this.setState({ values: this.props.defaultValues, restoreDefaultValuesModalIsOpen: false });
  };

  private handleDeploy = async (e: React.FormEvent<HTMLFormElement>) => {
    e.preventDefault();
    // Clean possible previous errors
    this.setState({ error: undefined });
    const { values } = this.state;
    let resource: IResource = {} as any;
    try {
      resource = yaml.safeLoad(values);
    } catch (e) {
      this.setState({
        error: new UnprocessableEntity(`Unable to parse the given YAML. Got: ${e.message}`),
      });
      return;
    }
    if (!resource.apiVersion) {
      this.setState({
        error: new UnprocessableEntity(
          "Unable parse the resource. Make sure it contains a valid apiVersion",
        ),
      });
      return;
    }
    const resourceName = get(resource, "metadata.name");
    this.setState({ submittedResourceName: resourceName });
    this.props.handleDeploy(resource);
  };
}

export default DeploymentFormBody;
