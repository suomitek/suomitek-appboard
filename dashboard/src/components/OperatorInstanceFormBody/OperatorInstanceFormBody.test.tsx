import { mount, shallow } from "enzyme";
import * as React from "react";
import Modal from "react-modal";
import { Tabs } from "react-tabs";
import OperatorInstanceFormBody from ".";
import itBehavesLike from "../../shared/specs";
import { ConflictError } from "../../shared/types";
import { ErrorSelector } from "../ErrorAlert";
import UnexpectedErrorPage from "../ErrorAlert/UnexpectedErrorAlert";
import { IOperatorInstanceFormProps } from "./OperatorInstanceFormBody";

const defaultProps: IOperatorInstanceFormProps = {
  csvName: "foo",
  isFetching: false,
  namespace: "kubeapps",
  handleDeploy: jest.fn(),
  defaultValues: "",
  errors: {},
};

itBehavesLike("aLoadingComponent", {
  component: OperatorInstanceFormBody,
  props: { ...defaultProps, isFetching: true },
});

it("set default values", () => {
  const wrapper = shallow(<OperatorInstanceFormBody {...defaultProps} />);
  wrapper.setProps({ defaultValues: "foo" });
  expect(wrapper.state()).toMatchObject({
    values: "foo",
  });
});

it("renders an error if there is some error fetching", () => {
  const wrapper = shallow(
    <OperatorInstanceFormBody {...defaultProps} errors={{ fetch: new Error("Boom!") }} />,
  );
  expect(wrapper.find(ErrorSelector)).toExist();
});

it("renders an error if the namespace is _all", () => {
  const wrapper = shallow(<OperatorInstanceFormBody {...defaultProps} namespace="_all" />);
  expect(wrapper.find(UnexpectedErrorPage)).toExist();
});

it("renders an error if the creation failed", () => {
  const wrapper = shallow(<OperatorInstanceFormBody {...defaultProps} />);
  wrapper.setProps({ errors: { create: new ConflictError() } });
  expect(wrapper.find(ErrorSelector)).toExist();
  expect(wrapper.find(Tabs)).toExist();
});

it("restores the default values", async () => {
  const wrapper = mount(<OperatorInstanceFormBody {...defaultProps} />);
  Modal.setAppElement(document.createElement("div"));
  wrapper.setProps({ defaultValues: "foo" });
  wrapper.setState({ values: "not-foo" });
  const restoreButton = wrapper.find("button").filterWhere(b => b.text() === "Restore Defaults");
  restoreButton.simulate("click");

  const restoreConfirmButton = wrapper.find("button").filterWhere(b => b.text() === "Restore");
  restoreConfirmButton.simulate("click");

  const { values } = wrapper.state() as any;
  expect(values).toEqual("foo");
});

it("should submit the form", () => {
  const handleDeploy = jest.fn();
  const wrapper = shallow(
    <OperatorInstanceFormBody {...defaultProps} handleDeploy={handleDeploy} />,
  );

  const values = "apiVersion: v1\nmetadata:\n  name: foo";
  wrapper.setState({ values });
  const form = wrapper.find("form");
  form.simulate("submit", { preventDefault: jest.fn() });

  const resource = {
    apiVersion: "v1",
    metadata: {
      name: "foo",
    },
  };
  expect(handleDeploy).toHaveBeenCalledWith(resource);
});

it("should catch a syntax error in the form", () => {
  const handleDeploy = jest.fn();
  const wrapper = shallow(
    <OperatorInstanceFormBody {...defaultProps} handleDeploy={handleDeploy} />,
  );

  const values = "metadata: invalid!\n  name: foo";
  wrapper.setState({ values });
  const form = wrapper.find("form");
  form.simulate("submit", { preventDefault: jest.fn() });

  expect(
    wrapper
      .find(ErrorSelector)
      .dive()
      .dive()
      .text(),
  ).toContain("Unable to parse the given YAML. Got: bad indentation");
  expect(handleDeploy).not.toHaveBeenCalled();
});

it("should throw an eror if the element doesn't contain an apiVersion", () => {
  const handleDeploy = jest.fn();
  const wrapper = shallow(
    <OperatorInstanceFormBody {...defaultProps} handleDeploy={handleDeploy} />,
  );

  const values = "metadata:\nname: foo";
  wrapper.setState({ values });
  const form = wrapper.find("form");
  form.simulate("submit", { preventDefault: jest.fn() });

  expect(
    wrapper
      .find(ErrorSelector)
      .dive()
      .dive()
      .text(),
  ).toContain("Unable parse the resource. Make sure it contains a valid apiVersion");
  expect(handleDeploy).not.toHaveBeenCalled();
});
