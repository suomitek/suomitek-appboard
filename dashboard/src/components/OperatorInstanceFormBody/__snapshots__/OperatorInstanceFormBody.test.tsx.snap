// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loading spinner matches the snapshot 1`] = `
<LoadingWrapper
  loaded={false}
  type={0}
/>
`;
